src/main/java/hu/bme/aut/temalab/temalaborvehicleleasing/service/CustomerService.java

```diff
@@ -73,4 +77,27 @@ public void giveOutBonusPoints(int amount, int topHowMany) {
 			customerRepository.save(c);
 		});
 	}
+	
+	/**
+	 * Finds inactive customers in the database. A customer is inactive if his/her last rental's start date is more than 2 years behind than the current date
+	 * 
+	 * @return a collection of inactive customers
+	 */
+	public Collection<Customer> findInactiveCustomers() {
+		
+		List<Customer> customers = customerRepository.findAll();
+		List<Customer> inactiveCustomers = new ArrayList<>();
+		
+		List<Rental> rentals;
+		
+		for(Customer c : customers) {
+			rentals = List.copyOf(rentalRepository.findByCustomer(c));
+			
+			if(rentals != null && !rentals.isEmpty()) {
+				if(ChronoUnit.YEARS.between(rentals.get(rentals.size() - 1).getStartDate(), LocalDate.now()) > 2) inactiveCustomers.add(c);
```

Comment:
A rentals rendezve van időpont alapján vagy honnét tudjuk, hogy az utolsó rental az?

Reply:
Hát ugye úgy van, hogy ha egy customer kibérel egy járművet, akkor a rentals tömbjébe bekerül egy rental. Ezáltal rentals tömb a startDate alapján rendezett lesz (növekvő).

Reply:
Igen, de akkor figyelni kell rá, hogy a rentals több milyen sorendbe kerülnek be az adatok. Pl.: Valaki plusz adatott kézzel bele rak, aminek korábbi a dátuma és már nincs benne a 2 évben akkor már inaktív lesz az adot customer

Reply:
Pl..: ha kipróbáloda mockito tesztnél, hogy az utolsó 2 rentals.add-ot megcseréled már elrontja a tesztett... Mivel ott hiába nem régi a bérlés nem azt nézi hanem a régebbit

Reply:
Yap, ez is igaz, igazából szerintem maradhat így max ha gond lesz vele később módosítunk rajta